source/extensions/filters/http/aws_lambda/sts_fetcher.cc

```diff
@@ -90,6 +90,8 @@ class StsFetcherImpl : public StsFetcher,
     // Chained assumption specifics 
     const std::string body = fmt::format(StsChainedFormatString, role_arn, now);
     message->body().add(body);
+    // this call resets the sha as nothing should be added prior to this.
```

Comment:
if nothing should be added, do we want an ASSERT here?

Reply:
Sure that was more of a paranoia thing on the filter. I will check if its even needed.

Reply:
is the casewhere it is reused and we need to call the function is that in a different place?

Reply:
This is the case where we are intentionally reusing it to avoid having to recreate it per run. 
That being said I am starting to wonder if we should stop having it as a field on the fetcher at all but rather should just have it live its full lifecycle within this function.